File: Src/Data/Entities/RefreshToken.cs
namespace MusicPlayerBackend.Data.Entities;

public sealed record RefreshToken : EntityBase
{
    public Guid UserId { get; set; }
    public Guid Jti { get; set; }
    public Guid Token { get; set; }
    public DateTimeOffset ValidDue { get; set; }
    public bool Revoked { get; set; }

    public User User { get; set; } = null!;
}

File: Src/Persistence/Entities/RefreshToken.fs
namespace MusicPlayerBackend.Persistence.Entities

open System

type RefreshTokenId = Guid
type Jti = Guid

type RefreshToken = {
    Id: RefreshTokenId
    Jti: Jti
    Token: Guid
    Revoked: bool
    ValidDue: DateTimeOffset
    CreatedAt: DateTimeOffset
    UpdatedAt: DateTimeOffset voption
}
